src/parser/hays.py

```diff
@@ -11,325 +10,303 @@
-    #                        last_name, first_name = original_name.split(", ")
-    #                        judicial_officer = f"{first_name} {last_name}"
+            return (disposition_rows, other_event_rows)
+        except Exception as e:
+            logger.info(f"Error formatting events and orders of the court: {e}")
+            return ([], [])
+        
+    def get_disposition_information(self, row, dispositions, case_data, table, county, case_soup, logger) -> List[Dict]:
+        try:
+            if not row:
+                logger.info(f"No dispositions to process.")
+                return dispositions 
+            
+            if len(row) >= 5:
+                # Extract judicial officer if present
+                judicial_officer = ""
+                if len(row[2]) > 18 and row[2].startswith("(Judicial Officer:"):
+                    judicial_officer = row[2][18:-1].strip()
+                
+                # Create a disposition entry
+                disposition = {
+                    "date": row[0],
+                    "event": row[1],
+                    "judicial officer": judicial_officer,
+                    "details": []
+                }
 
-                        disposition = {
-                            "date": row[0],
-                            "event": row[1],
-                            "judicial officer": judicial_officer,
-                            "details": []
-                        }
+                # Check if this row is a disposition
+                if row[1].lower() in ["disposition", "amended disposition", "deferred adjudication", "punishment hearing"]:
+                    details = {
+                        "charge": row[3],
+                        "outcome": row[4]
+                    }
+                    if len(row) > 5:
+                        details["additional_info"] = row[5:]
+                    disposition["details"].append(details)
+                    dispositions.append(disposition)
+                    dispositions.reverse()
+                else:
+                    logger.info("Row is not a disposition: %s", row)
 
-                        # Check if the event is "Disposition"
-                        if row[1].lower() == "disposition":
-                            details = {
-                                "charge": row[3],
-                                "outcome": row[4]
-                            }
-                            if len(row) > 5:
-                                details["additional_info"] = row[5:]
-                            disposition["details"].append(details)
-                            dispositions.append(disposition)
+            return dispositions
+        except Exception as e:
+            logger.info(f"Error getting disposition information: {e}")
+            return dispositions
+        
+    def parser_hays(self, county: str, case_number: str, logger, case_soup: BeautifulSoup) -> Dict[str, Dict]:
```

Comment:
Love the formatting of this method and how it's all come together section by section.